src/store/Controller.cc

```diff
@@ -355,6 +355,21 @@ Store::Controller::allowSharing(StoreEntry &entry, const cache_key *key)
         const bool found = anchorToCache(entry, inSync);
         if (found && !inSync)
             throw TexcHere("cannot sync");
+        if (!found) {
+            if (transients->hasWriter(entry)) {
+                Transients::EntryStatus entryStatus;
+                transients->status(entry, entryStatus);
+                if (!entryStatus.collapsed) {
```

Comment:
Why are we checking flags in shared memory rather than calling entry.hittingRequiresCollapsing()? For the caller, it is that latter flag that makes all the difference, right? AFAICT, unknown bugs notwithstanding, Transients::get() should have set that flag for these `!found` StoreEntries.